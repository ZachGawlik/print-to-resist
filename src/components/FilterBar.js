import React from 'react';
import PropTypes from 'prop-types';
import Tag from './Tag';
import { COLOR_OPTIONS } from '../constants';

class FilterBar extends React.Component {
  setColorFilterOption = event => {
    this.props.setColorFilter(event.target.value);
  };

  clearTagFilter = () => {
    this.props.setTagFilter(null);
  };

  render() {
    const { galleryFilters, toggleActiveOnly } = this.props;
    return (
      <div>
        <ul>
          <li>
            <label>
              Show only upcoming events {' '}
              <input
                type="checkbox"
                checked={galleryFilters.activeOnly}
                onChange={toggleActiveOnly}
              />
            </label>
          </li>
          <li>
            <label>
              To be printed in: {' '}
              <select
                value={galleryFilters.colorOption || ''}
                onChange={this.setColorFilterOption}
              >
                <option value="">All</option>
                <option value={COLOR_OPTIONS.color}>
                  {COLOR_OPTIONS.color}
                </option>
                <option value={COLOR_OPTIONS.bw}>{COLOR_OPTIONS.bw}</option>
              </select>
            </label>
          </li>
        </ul>
        <p>
          Showing {galleryFilters.activeOnly ? 'active' : 'all'}
          {galleryFilters.colorOption &&
            <span>{' '}to be printed in {galleryFilters.colorOption}</span>}
          {galleryFilters.tag &&
            <span>
              {' '}with tag{' '}
              <Tag onClick={this.clearTagFilter} name={galleryFilters.tag} />
            </span>}
        </p>
      </div>
    );
  }
}

FilterBar.propTypes = {
  galleryFilters: PropTypes.shape({
    colorOption: PropTypes.string.isRequired,
    activeOnly: PropTypes.bool.isRequired,
    tag: PropTypes.string
  }).isRequired,
  setColorFilter: PropTypes.func.isRequired,
  setTagFilter: PropTypes.func.isRequired,
  toggleActiveOnly: PropTypes.func.isRequired
};

export default FilterBar;
