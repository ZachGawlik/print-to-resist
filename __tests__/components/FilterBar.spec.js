import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FilterBar from '../../src/components/FilterBar';

describe('FilterBar', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <FilterBar
        galleryFilters={{
          activeOnly: false,
          colorOption: '',
          tag: null
        }}
        setColorFilter={jest.fn()}
        setTagFilter={jest.fn()}
        toggleActiveOnly={jest.fn()}
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
