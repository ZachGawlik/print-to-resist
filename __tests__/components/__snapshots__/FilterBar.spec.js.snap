// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar renders as expected 1`] = `
<div>
  <ul>
    <li>
      <label>
        Show only upcoming events 
         
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </label>
    </li>
    <li>
      <label>
        To be printed in: 
         
        <select
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            All
          </option>
          <option
            value="Color"
          >
            Color
          </option>
          <option
            value="Black and White"
          >
            Black and White
          </option>
        </select>
      </label>
    </li>
  </ul>
  <p>
    Showing 
    all
    
  </p>
</div>
`;
